### PR TITLE
feat(perf): Add key transasaction as a pseudo column

### PR DESCRIPTION
**Pull request by Zylphrex**
This includes the necessary changes in order to remove the `By Key Transaction`
pill that exists in the performance landing page today. This change introduces
a new pseudo-column that will resolve to whether or not a transaction is a key
transaction. The resulting value from snuba will be a number, making it possible
to sort by key transactions.

### CODE REVIEW

**Review by dashed — APPROVED** [review]

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -1250,25 +1272,117 @@ def get_filter(query=None, params=None):
...
+        self.name = name
+        self.alias = alias
+        self.expression = expression
+        self.expression_fn = expression_fn
+        self.result_type = result_type
+
+    def get_expression(self, params):
+        if isinstance(self.expression, (list, tuple)):
+            expression = deepcopy(self.expression)
+            return expression
```
Comment (markstory) [inline src/sentry/api/event_search.py:1286]:
```suggestion
            return deepcopy(self.expression)
```

🏌️

File: src/sentry/api/event_search.py [hunk 2]
```diff
@@ -1250,25 +1272,117 @@ def get_filter(query=None, params=None):
...
+        elif self.expression_fn is not None:
+            expression = self.expression_fn(params)
+            return expression
```
Comment (markstory) [inline src/sentry/api/event_search.py:1289]:
```suggestion
            return self.expression_fn(params)
```
🏌️

File: src/sentry/api/event_search.py [hunk 3]
```diff
@@ -1250,25 +1272,117 @@ def get_filter(query=None, params=None):
...
+# This function will only ever be called with 1 set of parameters in the lifetime of a
+# single request. So cache its results to avoid doing more lookups than necessary.
+@functools.lru_cache(maxsize=1)
+def key_transaction_expression(user_id, organization_id, project_ids):
+    if user_id is None or organization_id is None or project_ids is None:
+        raise InvalidSearchQuery("Missing necessary meta for key transaction field.")
+
+    key_transactions = (
+        KeyTransaction.objects.filter(
+            owner__id=user_id, organization__id=organization_id, project__id__in=project_ids,
```
Comment (markstory) [inline src/sentry/api/event_search.py:1317]:
```suggestion
            owner_id=user_id, organization_id=organization_id, project_id__in=project_ids,
```

Using the `__` will likely do joins as you're setting conditions on the associations. Using column names should avoid that.

File: src/sentry/api/event_search.py [hunk 4]
```diff
@@ -1250,25 +1272,117 @@ def get_filter(query=None, params=None):
...
+        )
+        .order_by("transaction", "project__id")
+        .values("project__id", "transaction")
```
Comment (markstory) [inline src/sentry/api/event_search.py:1320]:
```suggestion
        .order_by("transaction", "project_id")
        .values("project_id", "transaction")
```

File: src/sentry/api/event_search.py [hunk 5]
```diff
@@ -1250,25 +1272,117 @@ def get_filter(query=None, params=None):
...
+    return [
+        "has",
+        [
+            [
+                "array",
+                [
+                    [
+                        "tuple",
+                        [
+                            ["toUInt64", [transaction["project__id"]]],
```
Comment (markstory) [inline src/sentry/api/event_search.py:1336]:
```suggestion
                            ["toUInt64", [transaction["project_id"]]],
```

Comment (markstory) [inline src/sentry/api/event_search.py:1336]:
```suggestion
                            ["toUInt64", [transaction["project_id"]]],
```

File: src/sentry/api/event_search.py [hunk 6]
```diff
@@ -1250,25 +1272,117 @@ def get_filter(query=None, params=None):
...
-    USER_DISPLAY_ALIAS: {
-        "expression": ["coalesce", ["user.email", "user.username", "user.ip"]],
-        "field": ["coalesce", ["user.email", "user.username", "user.ip"], USER_DISPLAY_ALIAS],
-        "column_alias": USER_DISPLAY_ALIAS,
-    },
+    field.name: field
+    for field in [
+        PseudoField("project", "project.id"),
+        PseudoField("issue", "issue.id"),
+        PseudoField(ERROR_UNHANDLED_ALIAS, ERROR_UNHANDLED_ALIAS, expression=["notHandled", []]),
```
Comment (markstory) [inline src/sentry/api/event_search.py:1356]: Nice :clap:

File: src/sentry/api/event_search.py [hunk 7]
```diff
@@ -866,6 +869,19 @@ def convert_search_filter_to_snuba_query(search_filter, key=None):
...
             "Invalid value for error.handled condition. Accepted values are 1, 0"
         )
+    elif name == KEY_TRANSACTION_ALIAS:
+        key_transaction_expr = FIELD_ALIASES[KEY_TRANSACTION_ALIAS].get_expression(params)
+
+        if search_filter.value.raw_value == "":
+            operator = "!=" if search_filter.operator == "!=" else "="
+            return [key_transaction_expr, operator, 0]
+        if value in ("1", 1):
+            return [key_transaction_expr, "!=", 0]
```
Comment (wmak) [inline src/sentry/api/event_search.py:879]: nit:
```suggestion
            return [key_transaction_expr, "=", 1]
```

**Review by wmak — APPROVED** [review]
Nice

**Review by markstory — APPROVED** [review]

**Pull request merged**